aea/helpers/dialogue/base.py

```diff
@@ -738,20 +757,17 @@ def __init__(
         self._agent_address = agent_address
         self._dialogue_stats = DialogueStats(end_states)
 
-        if message_class is not None:
-            assert issubclass(message_class, Message)
+        assert issubclass(message_class, Message)
         self._message_class = message_class
 
-        if dialogue_class is not None:
-            assert issubclass(dialogue_class, Dialogue)
+        assert issubclass(dialogue_class, Dialogue)
         self._dialogue_class = dialogue_class
 
-        if role_from_first_message is not None:
-            self._role_from_first_message = role_from_first_message
-        else:
-            self._role_from_first_message = (
-                self.role_from_first_message
-            )  # pragma: no cover
+        sig = signature(role_from_first_message)
+        assert len(sig.parameters.keys()) == 1
+        assert list(sig.parameters.values())[0].annotation == Message
+        assert sig.return_annotation == Dialogue.Role
```

Comment:
Please provide helpful error messages for failure mode; same above in in lines 760, 763

Reply:
Done